src/algebra/algebra.jl

```diff
@@ -56,6 +83,21 @@ function dcontract(A::AbstractArray{T1, 3}, B::AbstractArray{T2, 3}) where {T1,
     return C
 end
 
+"""
+Tensors double contraction between fourth order tensor and second order tensor
+
+# Implementation
+A[i,j,k,l] : B[m,n] = C[i,j]
+such as
+C[i,j] = A[i,j,k,l] * B[k,l] (Einstein sum)
+"""
+function dcontract(A::AbstractArray{T1, 4}, B::AbstractArray{T2, 2}) where {T1, T2}
+    sA = size(A)
+    C = zeros(promote_type(eltype(A), eltype(B)), sA[1], sA[2])
+    C = [sum(A[i, j, :, :] .* B) for i in 1:sA[1], j in 1:sA[2]]
```

Comment:
can you replace `A[i,j,:,:]` by `view(A,i,j,:,:)` to avoid slice allocation ? (only for non static case)